pennylane/tape/qnode.py

```diff
@@ -515,6 +515,13 @@ def __call__(self, *args, **kwargs):
         # execute the tape
         res = self.qtape.execute(device=self.device)
 
+        # FIX: If the qnode swapped the device, increase the num_execution value on the original device.
+        # In the long run, we should make sure that the user's device is the one
+        # actually run so she has full control. This could be done by changing the class
+        # of the user's device before and after executing the tape.
+        if self.device != self._original_device:
+            self._original_device._num_executions += 1  # pylint: disable=protected-access

```

Comment:
This might sound crazy, but will `self.qtape.execute(device=self.device)` always result in just 1 additional execution?

Or is this a case where there could be more than 1 under the scenes?

I suppose I am being crazy

Reply:
I was wondering too. In future this may very well happen, but I think it does not happen at the moment. More reason to find a better fix one day.

Reply:
In that case you could do 

```python
diff_executions = self.device.num_executions
res = self.qtape.execute(device=self.device)
diff_executions = self.device.num_executions - diff_executions
if self.device is not self._original_device:
    self._original_device._num_executions += diff_executions
```